File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
       
        int maxSize=0;
        int i=0, j=0;
        
         set<char> st;
        while(j<s.size()){
            if(st.count(s[j]) == 0){
                maxSize = max(maxSize, j-i+1);
                st.insert(s[j++]);
            }
            else{
                st.erase(s[i++]);
            }
        }
        return maxSize;
    }
};

File: 3-longest-substring-without-repeating-characters/3-longest-substring-without-repeating-characters.java
class Solution {
    public int lengthOfLongestSubstring(String s) {
        
        int i=0 ,j = 0; 
        int maxSub = 0;  
        
        Set<Character> set = new HashSet<>();
        while(j < s.length()){
            if(!set.contains(s.charAt(j))){
                set.add(s.charAt(j++));
                maxSub = Math.max(maxSub,set.size());
            }
            else{
                set.remove(s.charAt(i++));
            }
        }
        return maxSub;
    }
}